services/api/db/serializers.py

```diff
@@ -0,0 +1,109 @@
+from rest_framework import serializers
+from db.models import User
+from django.utils.encoding import smart_str, force_bytes, DjangoUnicodeDecodeError
+from django.utils.http import urlsafe_base64_decode, urlsafe_base64_encode
+from django.contrib.auth.tokens import PasswordResetTokenGenerator
+from django.core.validators import validate_email
+
+
+class SignUpEndPointSerializer(serializers.ModelSerializer):
+    password2 = serializers.CharField(style={'input_type': 'passowrd'}, write_only=True)
+    class Meta:
+        model = User
+        fields= ['email', 'name', 'password', 'password2']
+        extra_kwargs={
+            'password': {'write_only': True}
+        }
+
+    #validating if password entered are correct.
+    def validate(self, data):
+        email = data.get('email')
+        password, password2 = data.get('password'), data.get('password2')
+        try:
+            validate_email(email)
+        except:
+            raise serializers.ValidationError("Please provide a valid email address.")
+        
+        if password != password2:
+            raise serializers.ValidationError("Password and confirm password doesn't match.")
+        return data
+        
+    def create(self, validated_data):
+        return User.objects.create_user(**validated_data)
+    
+class SignInEndPointSerializer(serializers.ModelSerializer):
+    email = serializers.EmailField(max_length=255)
+    class Meta:
+        model = User
+        fields = ['email', 'password']
+
+class UserProfileSerialiser(serializers.ModelSerializer):
+    class Meta:
+        model = User
+        fields = ['id', 'email', 'name']
+
+class UserChangePasswordSerialiser(serializers.Serializer):
+
+    #new password type can be customised here.
+    password = serializers.CharField(max_length=255, style={'input_type' : 'passowrd'}, write_only=True)
+    password2 = serializers.CharField(max_length=255, style={'input_type' : 'passowrd'}, write_only=True)
+    class Meta:
+        fields = ['passowrd', 'password2']
+    
+    def validate(self, attrs):
```

Comment:
Method name is validate, but it also saves the user.

Reply:
added comment

Reply:
changed this structure